src/presentation/http/router/noteSettings.test.ts

```diff
@@ -0,0 +1,109 @@
+import { describe, test, expect } from 'vitest';
+
+import notes from '@tests/test-data/notes.json';
+import noteSettings from '@tests/test-data/notes-settings.json';
+
+describe('NoteSettings API', () => {
+  describe('GET /note-settings/:notePublicId ', () => {
+    test('Returns note settings by public id with 200 status', async () => {
+      const expectedStatus = 200;
+      const existingNotePublicId = 'Pq1T9vc23Q';
+
+      const expectedNoteSettings = {
+        'customHostname': 'codex.so',
+        'isPublic': true,
+        'id': 2,
+        'noteId': 2,
+      };
+
+      const response = await global.api?.fakeRequest({
+        method: 'GET',
+        url: `/note-settings/${existingNotePublicId}`,
+      });
+
+      expect(response?.statusCode).toBe(expectedStatus);
+
+      expect(response?.json()).toStrictEqual(expectedNoteSettings);
+    });
+  });
+
+  test('Returns 406 when note settings with specified note public id do not exist', async () => {
+    const expectedStatus = 406;
+    const nonexistentId = 'ishvm5qH84';
+
+    const response = await global.api?.fakeRequest({
+      method: 'GET',
+      url: `/note-settings/${nonexistentId}`,
+    });
+
+    expect(response?.statusCode).toBe(expectedStatus);
+
+    expect(response?.json()).toStrictEqual({ message: 'Note not found' });
+  });
+
+  test.each([
+    { id: 'mVz3iHuez',
+      expectedMessage: 'params/notePublicId must NOT have fewer than 10 characters' },
+
+    { id: 'cR8eqF1mFf0',
+      expectedMessage: 'params/notePublicId must NOT have more than 10 characters' },
+
+    { id: '+=*&*5%&&^&-',
+      expectedMessage: '\'/note-settings/+=*&*5%&&^&-\' is not a valid url component' },
+  ])
```

Comment:
which is more priority? I mean if the id has > 10 characters and some of them are incorrect?

Reply:
There are three test cases: if the length of id < 10, if the length > 10 and if if the characters are invalid